src/analysis/type_analysis.cpp

```diff
@@ -365,8 +365,22 @@ class BasicBlockTypePropagator : public ExprVisitor, public StmtVisitor {
                 return BOOL;
             case AST_LangPrimitive::LOCALS:
                 return DICT;
+            case AST_LangPrimitive::GET_ITER: {
+                CompilerType* operand = getType(node->args[0]);
+                const std::string& name = "__iter__";
+                CompilerType* attr_type = operand->getattrType(&name, true);
+                CompilerType* iter = attr_type->callType(ArgPassSpec(0), {}, NULL);
+                if (iter != UNKNOWN) {
+                    const std::string& has_next_str = "__hasnext__";
+                    CompilerType* attr_type = iter->getattrType(&has_next_str, true);
+                    if (attr_type == UNDEF)
+                        return UNKNOWN;
+                    return iter;
+                }
```

Comment:
What do you think about adding a `getiterType()` function to CompilerType's?  It would at least let us move this functionality near the corresponding `getiter` implementation.